plugins/catalog-backend/src/service/NextEntitiesCatalog.ts

```diff
@@ -209,7 +209,12 @@ export class NextEntitiesCatalog implements EntitiesCatalog {
       .delete();
   }
 
-  async entityAncestry(rootRef: string): Promise<EntityAncestryResponse> {
+  async entityAncestry(
+    rootRef: string,
+    permissionOptions?: {
+      isEntityAuthorized: (entity: Entity) => Promise<boolean>;
```

Comment:
Did you consider the possibility of fetching the entire ancestry of the entity and doing the filtering as a second step? Batch permission requests are more efficient after because all the entities are loaded in a single DB query, so it would likely be faster and (I believe) allow you to avoid permission things ending up in the base NextEntitiesCatalog.

Reply:
I did, but since we would want to filter out not just unauthorized entities, but also their parents (and recursively those parents' parents, etc), it adds the additional step of removing "all the way up the tree", so to speak. While certainly possible, I attempted to do this in a clean way and it quickly became evident that the alternative would be much cleaner. I didn't consider batch permission requests though, so maybe it's worth another shot even if just to compare.

Reply:
The median tree depth is very very small, if that's interesting in this context. Like, 2.

Reply:
Useful context, thanks @freben! My motivation here is mostly to find a way to organize this which avoids leaking permission concepts into the base NextEntitiesCatalog.

Reply:
Yep understood, just mentioning that nugget in passing of it might help some iterate-or-batch decisions :)

Reply:
Just pushed a new commit containing the "wrapper-changes-only" implementation (recursively filtering out unauthorized items). Significantly longer, but doesn't mess with the underlying `EntitiesCatalog` at all, and is a much cleaner change to the method signature. wdyt?